pkg/controller/jobframework/setup.go

```diff
@@ -109,20 +109,28 @@ func (m *integrationManager) setupControllers(ctx context.Context, mgr ctrl.Mana
 	})
 }
 
-func (m *integrationManager) setupControllerAndWebhook(mgr ctrl.Manager, name string, fwkNamePrefix string, cb IntegrationCallbacks, options Options, opts ...Option) error {
-	if err := cb.NewReconciler(
+func (m *integrationManager) setupControllerAndWebhook(ctx context.Context, mgr ctrl.Manager, name string, fwkNamePrefix string, cb IntegrationCallbacks, options Options, opts ...Option) error {
+	r, _ := cb.NewReconciler(
```

Comment:
I think we should use the error to fail fast rather than late. It makes debugging easier.

What about the structure like:

```golang
if r, err := vb.NewReconciler(); err != nil {
   return fmt.Errorf("%s: %w", fwkNamePrefix, err)
} else  if err := r.SetupWithManager(mgr); err != nil {
   return fmt.Errorf("%s: %w", fwkNamePrefix, err)
}
```
if if does not work, then indeed it is better to declare the error on top.

Reply:
it makes sense, but it indeed needs the variables to be declared at the top

Reply:
ack, so 
```
if r, err := vb.NewReconciler(); err != nil {
   return fmt.Errorf("%s: %w", fwkNamePrefix, err)
} else  if err := r.SetupWithManager(mgr); err != nil {
   return fmt.Errorf("%s: %w", fwkNamePrefix, err)
}
```
does not compile?

Reply:
Actually, I think the new version with declaring outside is also ok, please keep.

Reply:
Sorry, for the confusion. 

I just wanted to say that I'm happy with both versions, my comment was to explore options.

I'm ok to keep as is.